scripts/deploy.backend.sh

```diff
@@ -4,28 +4,74 @@ case $ENV in
   "staging")
     ECDSA_KEY_NAME="test_key_1"
     WALLET="cvthj-wyaaa-aaaad-aaaaq-cai"
+    # For security reasons, mainnet root key will be hardcoded in the backend canister.
+    ic_root_key_der="null"
     ;;
   "ic")
     ECDSA_KEY_NAME="key_1"
     WALLET="yit3i-lyaaa-aaaan-qeavq-cai"
+    # For security reasons, mainnet root key will be hardcoded in the backend canister.
+    ic_root_key_der="null"
     ;;
   *)
     ECDSA_KEY_NAME="dfx_test_key"
+    # At the time of writing dfx outputs incorrect JSON with dfx ping (commas between object
+    # entries are missing).
```

Comment:
It _seems_ to work with dfx 0.20.0, as defined in dfx.json, for mainnet and localhost:
```
$ dfx ping ic | jq -r '.root_key | reduce .[] as $item ("opt vec { "; "\(.) \($item):nat8;") + " }"'
opt vec {  48:nat8; 129:nat8; 130:nat8; 48:nat8; 29:nat8; 6:nat8; 13:nat8; 43:nat8; 6:nat8; 1:nat8; 4:nat8; 1:nat8; 130:nat8; 220:nat8; 124:nat8; 5:nat8; 3:nat8; 1:nat8; 2:nat8; 1:nat8; 6:nat8; 12:nat8; 43:nat8; 6:nat8; 1:nat8; 4:nat8; 1:nat8; 130:nat8; 220:nat8; 124:nat8; 5:nat8; 3:nat8; 2:nat8; 1:nat8; 3:nat8; 97:nat8; 0:nat8; 129:nat8; 76:nat8; 14:nat8; 110:nat8; 199:nat8; 31:nat8; 171:nat8; 88:nat8; 59:nat8; 8:nat8; 189:nat8; 129:nat8; 55:nat8; 60:nat8; 37:nat8; 92:nat8; 60:nat8; 55:nat8; 27:nat8; 46:nat8; 132:nat8; 134:nat8; 60:nat8; 152:nat8; 164:nat8; 241:nat8; 224:nat8; 139:nat8; 116:nat8; 35:nat8; 93:nat8; 20:nat8; 251:nat8; 93:nat8; 156:nat8; 12:nat8; 213:nat8; 70:nat8; 217:nat8; 104:nat8; 95:nat8; 145:nat8; 58:nat8; 12:nat8; 11:nat8; 44:nat8; 197:nat8; 52:nat8; 21:nat8; 131:nat8; 191:nat8; 75:nat8; 67:nat8; 146:nat8; 228:nat8; 103:nat8; 219:nat8; 150:nat8; 214:nat8; 91:nat8; 155:nat8; 180:nat8; 203:nat8; 113:nat8; 113:nat8; 18:nat8; 248:nat8; 71:nat8; 46:nat8; 13:nat8; 90:nat8; 77:nat8; 20:nat8; 80:nat8; 95:nat8; 253:nat8; 116:nat8; 132:nat8; 176:nat8; 18:nat8; 145:nat8; 9:nat8; 28:nat8; 95:nat8; 135:nat8; 185:nat8; 136:nat8; 131:nat8; 70:nat8; 63:nat8; 152:nat8; 9:nat8; 26:nat8; 11:nat8; 170:nat8; 174:nat8; }


$ dfx --version
dfx 0.20.0
```
But if you are getting errors in some cases, for sure, dfx needs to get fixed! :-)

Reply:
Thanks! I changed it. I copied it from another project.